src/dbl_linked_list.py

```diff
@@ -0,0 +1,125 @@
+        head:       Always the most recent item added to a Linked List.
+        _size:      The length of the Linked List or number of Nodes stored
+                    in the list.
+
+        Methods:
+
+        push(value):    Given a value, add a new Node object to a Linked List.
+
+        append(value):  Given a value, adds it at the end of the line.
+
+        pop():          Pop the head Node off the list, return the value.
+
+        shift():        will remove the last value from the line and return it.
+
+        remove(val):    Given a Node, remove that Node from the Linked List.  Else,
+                        raise a ValueError.
+        """
+
+    def __init__(self, maybe_an_iterable=None):
+        """Intialize a doubly linked List Object."""
+        self.head = None
+        self.tail = None
+        self._size = 0
+        if maybe_an_iterable:
+            try:
+                for value in maybe_an_iterable:
+                    self.push(value)
+                nodes = [node for node in self._iterate_from(self.head)]
+                self.tail = nodes[-1]
+            except TypeError:
+                self.push(maybe_an_iterable)
+
+    def push(self, value):
+        """Add a node at the beginning of the line."""
+        self.head = Dbl_Node(value, self.head)
+        self._size += 1
+        if self._size == 1:
+            self.tail = self.head
+        elif self._size == 2:
+            nodes = [node for node in self._iterate_from(self.head)]
+            self.tail = nodes[-1]
+            nodes[-1].prev = self.head
+        else:
+            self.head.nxt.prev = self.head
+
+    def pop(self):
+        """Pop the first value off the head of LL and return it."""
+        if self.head is None:
+            raise IndexError("Cannot pop from an empty Linked List.")
+        val = self.head.value
+        self.head = self.head.nxt
+        self.head.prev = None
+        self._size -= 1
+        return val
+
+    def shift(self):
+        """will remove the last value from the tail of the list and return it."""
+        if self.tail is None:
+            raise IndexError("Cannot shift (remove the last node in the line) from an empty Linked List.")
+        val = self.tail.value
+        self.tail.prev.nxt = None
+        self.tail = self.tail.prev
+        self._size -= 1
+        return val
+
+    def append(self, value):
+        """will append a new node at the end of the line and will set the new tail to be this node."""
+        new_node = Dbl_Node(value, None, self.tail)
+        nodes = [node for node in self._iterate_from(self.head)]
+        if nodes:
+            self.tail = new_node
+            nodes[-1].nxt = self.tail
+            self._size += 1
+        else:
+            self.head = new_node
+            self.tail = new_node
+            self._size += 1
+
+    def remove(self, value):
+        """Remove the given node from the LL."""
+        nodes = [node for node in self._iterate_from(self.head)]
```

Comment:
Are these next two lies really doing much? Do your really need to iterate 3 times over the entire doubly linked list, creating 2 copies of it along the way?

You should be able to perform the `remove` operation, including error handling, in just one sweep of the whole data structure.